### PR TITLE
Implement select iterable operation on table

### PR DESCRIPTION
**Pull request by Manuri**
## Purpose
Support select iterable operation on table. Select is similar to map operation except that it returns a table instead of an array. Fixes https://github.com/ballerina-lang/ballerina/issues/5217

## Automation tests
 - Unit tests 
   Added

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets/no? yes

## Samples
```ballerina
struct Person {
    int id;
    float salary;
    string name;
}

struct Salary {
    int id;
    float salary;
}

function testSelectWithTable () {
	
    table<Person> dt = {};
    Person p1 = {id:1, name:"A", salary:100};
    Person p2 = {id:2, name:"B", salary:1500};
    Person p3 = {id:3, name:"C", salary:970};

    dt.add(p1);
    dt.add(p2);
    dt.add(p3);
    table salaries = dt.select(getSalary);

    io:println(salaries);  

}

function getSalary(Person p)(Salary) {
    Salary s = {id:p.id, salary:p.salary};
    return s;
}
```

```bash
{data: [{id:1, salary:100.0}, {id:2, salary:200.0}, {id:3, salary:300.0}]}
```

## Test environment
JDK 8

### CODE REVIEW

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/expressions/lambda/IterableOperationsTests.java [hunk 1]
```diff
@@ -263,4 +269,40 @@ public void testInStatement() {
...
+    }
+
+    @Test
+    public void testSelectCompatibleLambdaInputOutput() {
+        BValue[] returns = BRunUtil.invoke(basic, "testSelectCompatibleLambdaInputOutput");
+        Assert.assertNotNull(returns);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertEquals(returns[0].stringValue(),
+                "{data: [{id:1, salary:100.0}, {id:2, salary:200.0}, {id:3, " + "salary:300.0}]}");
+    }
```
Comment (anupama-pathirage) [inline tests/ballerina-test/src/test/java/org/ballerinalang/test/expressions/lambda/IterableOperationsTests.java]: Shall we move these new tests into org.ballerinalang.test.types.table.TableIterationTest class.

File: tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/basic-iterable.bal [hunk 2]
```diff
@@ -111,6 +111,91 @@ function testBasicMap2 () (string[] values) {
...
+    return salaryTable;
+}
+
+function testSelectCompatibleLambdaInputOutput() (table) {
+    table<Employee> dt = createTable();
+
+    table<EmployeeSalary> salaryTable = dt.select(getSalaryCompatibleInputOutput);
+    return salaryTable;
+}
+
```
Comment (anupama-pathirage) [inline tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/basic-iterable.bal]: Shall we move these into table-iteration.bal

File: tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/iterable-negative.bal [hunk 3]
```diff
@@ -67,3 +67,75 @@ function test7(){
...
+    table<EmployeeSalary> t = dt.select(getSalaryIncompatibleOutput);
+    return t;
+}
+
+function test10() (table) {
+    table<Employee> dt = createTable();
+
+    table<EmployeeSalary> t = dt.select(getSalaryIncompatibleInputOutput);
+    return t;
+}
```
Comment (anupama-pathirage) [inline tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/iterable-negative.bal]: Shall we create a new file under table for table-iteration-negative.bal

File: tests/ballerina-test/src/test/java/org/ballerinalang/test/expressions/lambda/IterableOperationsTests.java (hunk 1 above)
Comment (Manuri) [reply tests/ballerina-test/src/test/java/org/ballerinalang/test/expressions/lambda/IterableOperationsTests.java]: Done

File: tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/basic-iterable.bal (hunk 2 above)
Comment (Manuri) [reply tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/basic-iterable.bal]: Done

File: tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/iterable-negative.bal (hunk 3 above)
Comment (Manuri) [reply tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/iterable-negative.bal]: Done

**Pull request merged**